runelite-client/src/main/java/net/runelite/client/ui/FontManager.java

```diff
@@ -36,37 +39,53 @@
 	private static final Font runescapeSmallFont;
 	private static final Font runescapeBoldFont;
 
+	// By using a 2way map we're not dependant on the toString() method so we have control over the font naming in the UI.
+	private static final HashMap<String, Font> fontLookupMap;
+	// By using a linkedHashMap we decide the order in which the fonts a re represented which allows more intuitive ordering, i couldn't get ordering to swork with guave bimaps
+	private static final LinkedHashMap<Font, String> fontMap;
+
 	static
 	{
 		GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
 
 		try
 		{
-			Font font = Font.createFont(Font.TRUETYPE_FONT,
+			runescapeFont = Font.createFont(Font.TRUETYPE_FONT,
 				FontManager.class.getResourceAsStream("runescape.ttf"))
 				.deriveFont(Font.PLAIN, 16);
-			ge.registerFont(font);
-
-			runescapeFont = StyleContext.getDefaultStyleContext()
-					.getFont(font.getName(), Font.PLAIN, 16);
-			ge.registerFont(runescapeFont);
 
-			Font smallFont = Font.createFont(Font.TRUETYPE_FONT,
+			runescapeSmallFont = Font.createFont(Font.TRUETYPE_FONT,
 				FontManager.class.getResourceAsStream("runescape_small.ttf"))
 				.deriveFont(Font.PLAIN, 16);
-			ge.registerFont(smallFont);
-
-			runescapeSmallFont = StyleContext.getDefaultStyleContext()
-					.getFont(smallFont.getName(), Font.PLAIN, 16);
-			ge.registerFont(runescapeSmallFont);
 
-			Font boldFont = Font.createFont(Font.TRUETYPE_FONT,
+			runescapeBoldFont = Font.createFont(Font.TRUETYPE_FONT,
 					FontManager.class.getResourceAsStream("runescape_bold.ttf"))
 					.deriveFont(Font.PLAIN, 16);
-			ge.registerFont(boldFont);
 
-			runescapeBoldFont = StyleContext.getDefaultStyleContext()
-					.getFont(boldFont.getName(), Font.PLAIN, 16);
+			fontMap = new LinkedHashMap<>();
+			fontLookupMap = new HashMap<>();
+
+			fontMap.put(runescapeSmallFont, "rs small");
+			fontMap.put(runescapeFont, "rs default");
+			fontMap.put(runescapeBoldFont, "rs bold");
+			fontMap.forEach((k, v) -> fontLookupMap.put(v, k));
+
+			// Get all available fonts on the system
+			Font[] availableFonts = ge.getAllFonts();
+			//build bidirectional map
+			Arrays.stream(availableFonts).sorted(Comparator.comparing(Font::getFontName)).forEach(f ->
+			{
+				if (!fontLookupMap.containsKey(f.getFontName()) && !fontMap.containsKey(f))
+				{
+					// I believe font size 11 or 12 is for many fonts about the same size as font size 16 for the rs fonts
+					f = f.deriveFont(11f);
```

Comment:
you should never use uneven size as default, mainly when there is no other option. Also i believe for a lot of monospace fonts they will be pretty small with this

Reply:
Can you give a source which explains why you never should? For example ms word starts with size 11 calibri for as long as i can remember. Also do you have an example for a monospace font? i realize 11 is not gonna work for every font but default size doesn't work for most fonts so do you have another suggestion?

Reply:
Because for monospaced fonts they tend to get pretty ugly when they are not using even sizes. E.g try Terminus with 11px and 12px

Reply:
I meant they can be blurry, its safer to use even number. Maybe 10 then? i dont think size-per-font woud be reasonable. Also, do you have any kind of fallback when font no longer exists in system? because that can easily happen.

Reply:
10 is too small for all. Settings should default back to runescape font but that's only when you open the runelite settings when the set font doesn't exit. There is already a size per font but you can't change it (every rs font is 16 while all other fonts are huge at that point).